src/pages/CardCreate/Card/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useMemo, useState } from 'react'
+import { CardFormProps } from '..'
+import Card from '../../../components/Card'
+import {
+  CardBaseFormProps,
+  CardType,
+  CardTypeAccordingToStartsWith,
+} from '../../../components/Form'
+import { useFormChangedState } from '../../../context/Form/hooks'
+
+type CardModelKeys = (keyof CardBaseFormProps)[]
+interface CreateCardProp {
+  formRef: React.RefObject<CardFormProps>
+  cardType: CardType | undefined
+  setCardType: React.Dispatch<React.SetStateAction<CardType | undefined>>
+  setModalIsOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+const CreateCard = ({
+  formRef,
+  cardType,
+  setCardType,
+  setModalIsOpen,
+}: CreateCardProp) => {
+  const state = useFormChangedState()
+  const [number1, setNumber1] = useState('')
+  const [number2, setNumber2] = useState('')
+  const [number3, setNumber3] = useState('')
+  const [number4, setNumber4] = useState('')
+
+  const [owner, setOwner] = useState('')
+
+  const [month, setMonth] = useState('')
+  const [year, setYear] = useState('')
+
+  useEffect(() => {
+    if (number1.length === 4 && number2.length === 4) {
+      const startNumber =
+        +number1[0] as keyof typeof CardTypeAccordingToStartsWith
+      setCardType(CardTypeAccordingToStartsWith[startNumber])
+    }
+  }, [number1, number2, setCardType])
+
+  const stateKeys = useMemo(() => Object.keys(state) as CardModelKeys, [state])
+
+  useEffect(() => {
+    const key = stateKeys.filter((key) => state[key] === true)[0]
+
+    switch (key) {
+      case 'number1':
+        setNumber1(formRef.current?.cardNumber1() ?? '')
+        break
+      case 'number2':
+        setNumber2(formRef.current?.cardNumber2() ?? '')
+        break
+      case 'number3':
+        setNumber3(formRef.current?.cardNumber3() ?? '')
+        break
+      case 'number4':
+        setNumber4(formRef.current?.cardNumber4() ?? '')
+        break
+      case 'expireAtMonth':
+        setMonth(formRef.current?.expiredAtMonth() ?? '')
+        break
+      case 'expireAtYear':
+        setYear(formRef.current?.expiredAtYear() ?? '')
+        break
+      case 'owner':
+        setOwner(formRef.current?.owner() ?? '')
+        break
+
+      default:
+        break
```

Comment:
```suggestion
 if(key === 'number1') setNumber1(formRef.current?.cardNumber1() ?? '');
if(key === 'number2') setNumber1(formRef.current?.cardNumber1() ?? '');
if(key === 'number3') setNumber1(formRef.current?.cardNumber1() ?? '');
if(key === 'number4') setNumber1(formRef.current?.cardNumber1() ?? '');
if(key === 'expiredAtMonth')  setMonth(formRef.current?.expiredAtMonth() ?? '')
if(key === 'expiredAtYear') setYear(formRef.current?.expiredAtYear() ?? '')
if(key === 'owner')  setOwner(formRef.current?.owner() ?? '')
```
state를 유지해야한다면 이부분은 if문으로 해도 괜찮지 않을까 생각하는데, 굳이 바꾸지 않아도 괜찮고요 (근데 바꿔보니 type 경고가 뜨는 부분이 있네요)